packages/datadog-instrumentations/src/prisma.js

```diff
@@ -0,0 +1,141 @@
+'use strict'
+
+const { isPromise } = require('util/types')
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+
+const primsaClientStartCH = channel('apm:prisma:client:start')
+const primsaClientFinishCH = channel('apm:prisma:client:finish')
+const prismaClientErrorCH = channel('apm:prisma:client:error')
+
+const prismaEngineStart = channel('apm:prisma:engine:start')
+
+const allowedClientSpanOperations = [
+  'operation',
+  'serialize',
+  'transaction'
+]
+class TracingHelper {
+  dbConfig = null
+  isEnabled () {
+    return true
+  }
+
+  // needs a sampled tracecontext to generate engine spans
+  getTraceParent (context) {
+    return '00-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-bbbbbbbbbbbbbbbb-01' // valid sampled traceparent
+  }
+
+  dispatchEngineSpans (spans) {
+    const roots = spans.filter((span) => span.parentId === null)
+    for (const root of roots) {
+      prismaEngineStart.publish({ engineSpan: root, allEngineSpans: spans, dbConfig: this.dbConfig })
+    }
```

Comment:
Nit: this prevents the extra array allocation:

```suggestion
    for (const span of spans) {
      if (span.parentId === null) {
        prismaEngineStart.publish({ engineSpan: span, allEngineSpans: spans, dbConfig: this.dbConfig })
      }
    }
```